virttest/qemu_vm.py

```diff
@@ -4220,14 +4220,17 @@ def _shell_reboot(session, timeout):
             _check_go_down = partial(_go_down, session, timeout)
         elif method == "system_reset":
             _reboot = self.system_reset
-            _check_go_down = partial(bool, True)
+            _check_go_down = None
         else:
             raise virt_vm.VMRebootError("Unknown reboot method: %s" % method)
-        # We consider QMPMonitor event as the most reliable method, always
-        # use it when available.
-        if isinstance(self.monitor, qemu_monitor.QMPMonitor):
-            _check_go_down = _go_down_qmp
-            self.monitor.clear_event("RESET")
+        if _check_go_down is None:
+            if isinstance(self.monitor, qemu_monitor.QMPMonitor):
```

Comment:
I understand, but it's too frequent for me. Having a switch is also something I considered but wanted to avoid, anyway it looks like the only option that would suit both of us. Let me add it (tomorrow).